win32/tests/threadpool_win32_int/threadpool_win32_int.c

```diff
@@ -896,96 +912,94 @@ static DWORD WINAPI chaos_thread_with_timers_func(LPVOID lpThreadParameter)
-                        if (threadpool_timer_start(chaos_test_data->threadpool, timer_start_delay, timer_period, work_function, (void*)&chaos_test_data->executed_timer_functions, &chaos_test_data->timers[which_timer_slot].timer) == 0)
-                        {
-                            InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STARTED);
-                        }
-                        else
-                        {
-                            InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_NONE);
-                        }
+                        InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STARTED);
+                    }
+                    else
+                    {
+                        InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_NONE);
                     }
                 }
-                (void)InterlockedDecrement(&chaos_test_data->timers_starting);
-                WakeByAddressSingle((PVOID)&chaos_test_data->timers_starting);
             }
-            break;
-        case 4:
+            (void)InterlockedDecrement(&chaos_test_data->timers_starting);
+            WakeByAddressSingle((PVOID)&chaos_test_data->timers_starting);
+        }
+        break;
+        case TEST_ACTION_DESTROY_TIMER:
             // Stop a timer
+        {
+            // Synchronize with close
+            (void)InterlockedIncrement(&chaos_test_data->timers_starting);
+            if (InterlockedAdd(&chaos_test_data->can_start_timers, 0) != 0)
             {
-                // Synchronize with close
-                (void)InterlockedIncrement(&chaos_test_data->timers_starting);
-                if (InterlockedAdd(&chaos_test_data->can_start_timers, 0) != 0)
+                int which_timer_slot = rand() * MAX_TIMER_COUNT / (RAND_MAX + 1);
+                if (InterlockedCompareExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STOPPING, TIMER_STATE_STARTED) == TIMER_STATE_STARTED)
                 {
-                    int which_timer_slot = rand() * MAX_TIMER_COUNT / (RAND_MAX + 1);
-                    if (InterlockedCompareExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STOPPING, TIMER_STATE_STARTED) == TIMER_STATE_STARTED)
-                    {
-                        threadpool_timer_destroy(chaos_test_data->timers[which_timer_slot].timer);
-                        chaos_test_data->timers[which_timer_slot].timer = NULL;
-                        InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_NONE);
-                    }
+                    threadpool_timer_destroy(chaos_test_data->timers[which_timer_slot].timer);
+                    //chaos_test_data->timers[which_timer_slot].timer = NULL;
```

Comment:
>//chaos_test_data->timers[which_timer_slot].timer = NULL; [](http://example.com/codeflow?start=20&length=57)

Commented out code detected.

Reply:
Either the code is not needed at all in which case it should be removed, or it is neededand then it should be uncommented

Reply:
yeah it's needed, uncomment the code